### PR TITLE
4461 add file asset service visibility selector

### PR DESCRIPTION
**Pull request by cstns**
## Description

add file asset service visibility selector and integrate it with the existing api
added a new key on the staticAsset api to include the current folder 
added the url column in the files list
added alerts

## Related Issue(s)
branched out of https://github.com/FlowFuse/flowfuse/pull/4473
closes https://github.com/FlowFuse/flowfuse/issues/4461

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by cstns** [comment]
Mobile is not yet optimized

File: forge/ee/routes/staticAssets/index.js [hunk 1]
```diff
@@ -69,6 +70,23 @@ module.exports = async function (app) {
                     }
                 }
             })
+
+            const parentDirectoryName = filePath.split('/').pop()
+            const parentDirectoryPath = filePath.split('/').slice(0, -1).join('/')
```
[the remark this answers is not in the document]
Comment (hardillb) [reply forge/ee/routes/staticAssets/index.js:75]: `/` is correct, the backend will handle it correctly, the frontend shouldn't care about the backend

File: frontend/src/components/file-browser/cells/FilePath.vue [hunk 2]
```diff
@@ -57,7 +57,7 @@ export default {
     },
     methods: {
         copyPath () {
-            navigator.clipboard.writeText(this.path)
+            navigator.clipboard.writeText('/' + this.path)
```
Comment (hardillb) [inline frontend/src/components/file-browser/cells/FilePath.vue:60]: Can we do this in what is displayed as well please

**Comment by hardillb** [comment]
image

We have double slashes in the URLs

**Comment by hardillb** [comment]
@cstns Sorry, I've just worked out why @joepavitt was using relative paths for files.

We can't use a standard `/data/storage/` on LocalFS builds because the storage directory is going to be something like `/opt/flowforge/var/projects/[project-id]/storage` and we don't have an easy way to get that path in the UI.

So please revert my suggestion to have the full path for the Folder Path and the File Path both in what's displayed and when copied to the clip board. The paths should also not start with a leading `/` (which might fix the URLs having double / in the last comment).

Again sorry for making you change that earlier

**Comment by cstns** [comment]
> Again sorry for making you change that earlier

No need to apologize, it's an easy revert.

Should I also revert the Base URL & File URL to relative paths or leave the fqdn as it is?

**Comment by hardillb** [comment]
No, leave the urls please

**Review by hardillb — APPROVED** [review]
I'm saying this is good

**Comment by hardillb** [comment]
But it has UI test failures so not merging

**Pull request merged**